src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +110,15 @@ public int Count
             }
         }
 
-        public Enumerator GetEnumerator()
+        public T this[int index]
         {
-            if (_array == null)
-                ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_NullArray);
-            Contract.EndContractBlock();
+            get { return _array[_offset + index]; }
```

Comment:
@jkotas, do you think argument validation should be added here to prevent people from accessing items that aren't within range of the segment?